### PR TITLE
Fix metacardLocation in TransformRequest

### PR DESCRIPTION
**Pull request by emmberk**
Before:
```
DEBUG 22 --- [nio-8080-exec-1] o.s.web.client.RestTemplate              : Writing [class TransformRequest {
    location: http://store:8080/dataset/b2958ded9ddf454f9bcdd300629eefa5
    mimeType: text/plain
    metacardLocation: http://store:8080/dataset/b2958ded9ddf454f9bcdd300629eefa5
}] with org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
```

After:
```
DEBUG 22 --- [nio-8080-exec-1] o.s.web.client.RestTemplate              : Writing [class TransformRequest {
    location: http://store:8080/dataset/d8621303bc904234aa433bd1a7928213
    mimeType: text/plain
    metacardLocation: http://store:8080/dataset/d8621303bc904234aa433bd1a7928213/metacard
}] with org.springframework.http.converter.json.MappingJackson2HttpMessageConverter
```

### CODE REVIEW

File: src/test/java/com/connexta/store/StoreITests.java [hunk 1]
```diff
@@ -450,43 +448,41 @@ public void testRetrieveIrmWhenS3IsEmpty() {
         is(HttpStatus.NOT_FOUND));
   }
 
-  @NotNull
-  private Matcher<URI> getRequestIsSuccessful() {
```
Comment (emmberk) [inline src/test/java/com/connexta/store/StoreITests.java]: This method was copied to the bottom of the class. It was not modified.

File: src/test/java/com/connexta/store/StoreITests.java [hunk 2]
```diff
@@ -450,43 +448,41 @@ public void testRetrieveIrmWhenS3IsEmpty() {
...
         .andExpect(jsonPath("$.mimeType").value(MediaType.TEXT_PLAIN_VALUE))
         .andExpect(
             request -> {
               final String metacardLocation =
-                  (new JsonPathExpectationsHelper("$.metacardLocation"))
+                  (String)
+                      new JsonPathExpectationsHelper("$.metacardLocation")
                           .evaluateJsonPath(
-                              ((MockClientHttpRequest) request).getBodyAsString(), String.class)
-                      + "/metacard";
```
Comment (emmberk) [inline src/test/java/com/connexta/store/StoreITests.java]: This line is why the test wasn't broken before.

**Comment by emmberk** [comment]
I heroed this locally.

File: src/main/java/com/connexta/store/config/StoreServiceConfiguration.java [hunk 3]
```diff
@@ -32,7 +31,6 @@ public StoreService storeService(
       @NotNull @Named("irmStorageAdaptor") final StorageAdaptor irmStorageAdapter,
       @NotNull @Named("metacardStorageAdaptor") final S3StorageAdaptor metacardStorageAdapter,
       @NotNull final IndexDatasetClient indexDatasetClient,
-      @NotNull final StatusService statusService,
```
Comment (mcalcote) [inline src/main/java/com/connexta/store/config/StoreServiceConfiguration.java]: ❓ This was added as part of @ahoffer's poller changes. It wouldn't break the transform request. Did you mean to remove it?

File: src/test/java/com/connexta/store/service/impl/StoreServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.MediaType;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@ExtendWith(MockitoExtension.class)
+class StoreServiceImplTest {
+
+  @Test
+  @Ignore("TODO")
+  void testRetrieveFile() {}
```
Comment (peterhuffer) [inline src/test/java/com/connexta/store/service/impl/StoreServiceImplTest.java]: ✏️ You can remove all the todos in this test class, I'm addressing them in my PR.

File: src/main/java/com/connexta/store/config/StoreServiceConfiguration.java (hunk 3 above)
Comment (emmberk) [reply src/main/java/com/connexta/store/config/StoreServiceConfiguration.java]: I didn't want to mock it in the unit tests. We can add it in once it's actually being used.

**Pull request merged**